src/graphql/data/nft/Details.ts

```diff
@@ -93,92 +91,85 @@ const detailsQuery = graphql`
   }
 `
 
-export function useLoadDetailsQuery(address?: string, tokenId?: string): void {
-  const [, loadQuery] = useQueryLoader(detailsQuery)
-  useEffect(() => {
-    if (address && tokenId) {
-      loadQuery({ address, tokenId })
-    }
-  }, [address, tokenId, loadQuery])
-}
-
-export function useDetailsQuery(address: string, tokenId: string): [GenieAsset, CollectionInfoForAsset] | undefined {
-  const queryData = useLazyLoadQuery<DetailsQuery>(
-    detailsQuery,
-    {
+export function useNftAssetDetails(
+  address: string,
+  tokenId: string
+): { data: [GenieAsset, CollectionInfoForAsset]; loading: boolean } {
+  const { data: queryData, loading } = useDetailsQuery({
+    variables: {
       address,
       tokenId,
     },
-    { fetchPolicy: 'store-or-network' }
-  )
+    fetchPolicy: 'cache-and-network',
+  })
 
-  const asset = queryData.nftAssets?.edges[0]?.node
+  const asset = queryData?.nftAssets?.edges[0]?.node as NonNullable<NftAsset>
   const collection = asset?.collection
   const listing = asset?.listings?.edges[0]?.node
   const ethPrice = parseEther(listing?.price?.value?.toString() ?? '0').toString()
 
-  return [
-    {
-      id: asset?.id,
-      address,
-      notForSale: asset?.listings === null,
-      collectionName: asset?.collection?.name ?? undefined,
-      collectionSymbol: asset?.collection?.image?.url ?? undefined,
-      imageUrl: asset?.image?.url ?? undefined,
-      animationUrl: asset?.animationUrl ?? undefined,
-      // todo: fix the back/frontend discrepency here and drop the any
-      marketplace: listing?.marketplace.toLowerCase() as any,
-      name: asset?.name ?? undefined,
-      priceInfo: {
-        ETHPrice: ethPrice,
-        baseAsset: 'ETH',
-        baseDecimals: '18',
-        basePrice: ethPrice,
-      },
-      susFlag: asset?.suspiciousFlag ?? undefined,
-      sellorders: asset?.listings?.edges.map((listingNode) => {
-        return {
-          ...listingNode.node,
-          protocolParameters: listingNode.node.protocolParameters
-            ? JSON.parse(listingNode.node.protocolParameters.toString())
-            : undefined,
-        } as SellOrder
-      }),
-      smallImageUrl: asset?.smallImage?.url ?? undefined,
-      tokenId,
-      tokenType: (asset?.collection?.nftContracts && asset?.collection.nftContracts[0]?.standard) as NftStandard,
-      collectionIsVerified: asset?.collection?.isVerified ?? undefined,
-      rarity: {
-        primaryProvider: 'Rarity Sniper', // TODO update when backend adds more providers
-        providers: asset?.rarities
-          ? asset?.rarities?.map((rarity) => {
-              return {
-                rank: rarity.rank ?? undefined,
-                score: rarity.score ?? undefined,
-                provider: 'Rarity Sniper',
-              }
-            })
-          : undefined,
+  return {
```

Comment:
have you had issues with this hook running more than required? it will re-render consumers because it's a new object every time